src/main/resources/Macros/Video20.xml

```diff
@@ -469,10 +469,9 @@ Compute the width and height.
 {{/html}}
 
 #else
-   #if ($xcontext.macro.params.attachment)
+   #if ("$!xcontext.macro.params.attachment" != '')
       ## read a video attachement
-      #if ($xcontext.macro.params.attachmentName != "")
-         #set($videoAttachmentName = $xcontext.macro.params.attachmentName)
+         #set($videoAttachmentName = $xcontext.macro.params.attachment)
```

Comment:
You have removed a parameter (`attachmentName`) and repurposed an existing parameter (`attachment`) of the macro *after* a release was already made with the previous macro signature.

This means that backwards compatibility is now broken between versions 1.6 (the previous version using the old parameters) and versions 1.7 (the just released one). So anyone using the 1.6 version will have their code fail when upgrading.

It`s very good that the documentation is updated with the recent changes!

What is missing now is to document the previously removed parameters (the backwards compatibility breakage) so that people will know what to do when upgrading. Basically, adding the old parameters in the list together with a note similar to "Removed in version 1.7. Use 'attachment' instead" (for `attachmentName`) and something similar to "Until 1.6: Boolean flag marking if the video should be displayed from an attachment. Specify the attachment's name using the 'attachmentName' parameter." (for `attachment`).

This extra documentation step is mandatory when making changes that break compatibility, but in the code the breakage should generally be avoided in minor versions and introduced in major versions, like 2.0 in your case. The practice is to deprecate things first and then drop them when jumping to a major version, so that users have time to adjust their code to the changes.

Of course, the decision between applying the deprecation process or directly dropping is left to the maintainers of the codebase, since it involves an increase in the maintainability cost, but the change needs to be acknowledge as breaking and needs to be documented.